src/gate_detect/src/GateDetector.cpp

```diff
@@ -111,8 +86,9 @@ GateCoordinates GateDetector::initialize(const cv::Mat matin) {
                     houghLinesThreshold_,
                     houghLinesMinLength_,
                     houghLinesMaxLineGap_);
+    Gate gate = Gate();
 
-    if (detectedLines.empty()) return gateCoordinates;
+    if (detectedLines.empty()) return gate;
```

Comment:
I'm confused, we return a gate if we couldn't find one? I would suggest looking at `std::optional` for a cleaner way to return in failure cases: https://en.cppreference.com/w/cpp/utility/optional

Reply:
Yeah that would be ideal, looks like optional is only included in c++17 from my research. I think a default zero gate may do for this?